src/crepe/system/InputSystem.cpp

```diff
@@ -0,0 +1,154 @@
+#include "../api/Button.h"
+#include "../api/EventManager.h"
+#include "ComponentManager.h"
+
+#include "system/InputSystem.h"
+
+using namespace crepe;
+
+void InputSystem::update() {
+	EventManager & event_mgr = EventManager::get_instance();
+	std::vector<SDLContext::EventData> event_list = SDLContext::get_instance().get_events();
+
+	for (const SDLContext::EventData & event : event_list) {
+		switch (event.event_type) {
+			case SDLContext::EventType::KEYDOWN: {
+				event_mgr.queue_event<KeyPressEvent>(KeyPressEvent{
+					.repeat = event.key_repeat,
+					.key = event.key,
+				});
+				break;
+			}
+			case SDLContext::EventType::KEYUP: {
+				event_mgr.queue_event<KeyReleaseEvent>(KeyReleaseEvent{
+					.key = event.key,
+				});
+				break;
+			}
+			case SDLContext::EventType::MOUSEDOWN: {
+				event_mgr.queue_event<MousePressEvent>(MousePressEvent{
+					.mouse_x = event.mouse_position.first,
+					.mouse_y = event.mouse_position.second,
+					.button = event.mouse_button,
+				});
+				last_mouse_down_position = event.mouse_position;
+				last_mouse_button = event.mouse_button;
+				break;
+			}
+			case SDLContext::EventType::MOUSEUP: {
+				MouseReleaseEvent mouse_release_event = MouseReleaseEvent{
+					.mouse_x = event.mouse_position.first,
+					.mouse_y = event.mouse_position.second,
+					.button = event.mouse_button,
+				};
+				event_mgr.queue_event<MouseReleaseEvent>(mouse_release_event);
```

Comment:
Het valt mij op dat je overal het event aanmaakt in de `event_mgr.queue_event<...>()` en hier maak je eerste een `MouseReleaseEvent`, om die daarna als argument mee te geven.
Het is niet fout, maar het viel mij gewoon op.